src/components/CustomerDetails/CollectCustomerDetailsScreen.tsx

```diff
@@ -204,6 +204,11 @@ const CollectCustomerDetailsScreen: FunctionComponent<NavigationFocusInjectedPro
   const onBarCodeScanned: BarCodeScannedCallback = (event) => {
     if (isFocused && isScanningEnabled && event.data) {
       onCheck(event.data);
+      if (event.type.includes("Code39")) {
+        onCheck(event.data);
+      } else {
+        onCheck(JSON.parse(event.data).passportId);
```

Comment:
Not sure if this looks better, and you can easily access properties you add to data in the future too!

```suggestion
        const { passportId } = JSON.parse(data);
        onCheck(passportId);
```